internal/bminventory/inventory.go

```diff
@@ -2436,10 +2436,8 @@ func (b *bareMetalInventory) updateOperatorsData(ctx context.Context, cluster *c
 	//At this point, if any operators are updated, retrigger auto-assign
 	//role calculation. This reset is needed because operators may affect
 	//the role assignment logic.
-	//There is no need to refresh the status here because the refresh
-	//occurs outside the function as part of the update cluster flow
 	if len(updateOLMOperators) > 0 || len(removedOLMOperators) > 0 {
-		if reset_err := b.resetAutoAssignRoles(ctx, db, cluster, log, false); reset_err != nil {
+		if _, reset_err := host.ResetAutoAssignRoles(cluster, db, log); reset_err != nil {
```

Comment:
I removed it from the cluster API because it needed as utility at the host monitor. So now it is a utility and needed to be on the host package because the host monitor cannot import the cluster package due to cyclic dependecies

Reply:
Q> "The whole role assignment operation should be part of cluster. Not hosts. If this is done, no need to update the current hosts after role is set. In addition, the way it is now, every selectRole operation needs a DB query to get all hosts, instead of one query to get all."
A> It can not be because the logic of selectRole depends on the *host* validations. Therefor I need to locate the logic at the host monitor. As far as I understand, I can not bring the host validation code into the cluster validation

Reply:
"The resetRoleAssignmentIfNotAllRolesAreSet operation can be done in a single DB operation instead of bringing data to the service, and then perform update." 
The can be done only if the query is done before the hosts are read from the DB and joining the cluster but then the read should be done separately.  Again if I could run the host validation within the cluster monitor and could be done but shouldn't they be seprarted?

Reply:
If the logic of selectRole depends on the host validations, maybe these validations should rely on functions that are in common.  This way, to select the role we will not have to run the refreshHost logic, but just the functions needed for role selection.

Reply:
There are 6 such validations and growing. That mean quite a rewrite. @filanov what do you think?